shed/event_streams.py

```diff
@@ -1073,81 +1073,95 @@ class BundleSingleStream(EventStream):
     def __init__(self, child, control, predicate_against=('start', 'stop'),
                  **kwargs):
         self.predicate_against = predicate_against
-        self.maxsize = kwargs.pop('maxsize', 100)
-        self.buffers = []
-        self.desc_start_map = {}
-        self.condition = Condition()
-        self.prior = ()
         self.control = control
-        self.start_count = 0
         if isinstance(control, int):
             EventStream.__init__(self, child=child)
             self.n_hdrs = control
-            self.predicate = lambda x: self.start_count == self.n_hdrs
+            self.predicate = lambda x, x2: self.start_count == self.n_hdrs
         elif callable(control):
             EventStream.__init__(self, child=child)
             self.n_hdrs = None
             self.predicate = control
         else:
             EventStream.__init__(self, children=(child, control))
             self.n_hdrs = None
-            self.predicate = lambda x: self.start_count == self.n_hdrs
+            self.predicate = lambda x, x2: self.start_count == self.n_hdrs
+
         self.generate_provenance()
-        self.emitted = {'start': False, 'descriptor': False}
-        self.start_docs = None
+        self.emitted = {'start': 0, 'descriptor': 0, 'event': 0, 'stop': 0}
+        self.predicate_docs = dict()
+        for name in predicate_against:
+            self.predicate_docs[name] = deque(maxlen=2)
+            self.predicate_docs[name].append(None)
+
+        self.issue_stop = False
 
     def update(self, x, who=None):
         return_values = []
         name, docs = self.curate_streams(x, False)
+        # only accepts first doc
```

Comment:
TODO : this should be added in docstring that subsequent docs ignored

Reply:
yes please!